src/mobile-token/use-token-toggle-details.ts

```diff
@@ -1,36 +1,45 @@
-import {useEffect, useState} from 'react';
 import {useMobileTokenContextState} from '@atb/mobile-token';
+import { useIsFocused } from '@react-navigation/native';
+import {useQuery} from '@tanstack/react-query';
 
-const useTokenToggleDetails = (shouldFetchTokenDetails: boolean) => {
-  const [shouldShowLoader, setShouldShowLoader] = useState<boolean>(false);
-  const [toggleLimit, setToggleLimit] = useState<number | undefined>();
-  const [maxToggleLimit, setMaxToggleLimit] = useState<number | undefined>();
+const ONE_MINUTE = 1000 * 60;
+export const GET_TOKEN_TOGGLE_DETAILS_QUERY_KEY = 'getTokenToggleDetails';
 
-  const {getTokenToggleDetails} = useMobileTokenContextState();
+const useTokenToggleDetails = () => {
+  const {getTokenToggleDetails, deviceInspectionStatus} = useMobileTokenContextState();
+  const screenHasFocus = useIsFocused();
+  const shouldFetchTokenDetails =
+    screenHasFocus && deviceInspectionStatus !== 'loading';
+  const queryResult = useQuery({
+    enabled: shouldFetchTokenDetails,
+    queryKey: [GET_TOKEN_TOGGLE_DETAILS_QUERY_KEY],
+    queryFn: async () => {
+      let toggleLimit: number | undefined;
+      let maxToggleLimit: number | undefined;
 
-  useEffect(() => {
-    (async function () {
-      if (shouldFetchTokenDetails) {
-        setShouldShowLoader(true);
-        const toggleToggleDetails = await getTokenToggleDetails();
-        if (toggleToggleDetails) {
-          const {toggleMaxLimit, toggledCount} = toggleToggleDetails;
-          if (toggleMaxLimit) {
-            if (toggleMaxLimit >= toggledCount) {
-              setToggleLimit(toggleMaxLimit - toggledCount);
-            } else {
-              //We can end up here if we decide to reduce max limit value in firestore
-              setToggleLimit(0);
-            }
+      const tokenToggleDetails = await getTokenToggleDetails();
+      if (tokenToggleDetails) {
+        const {toggleMaxLimit, toggledCount} = tokenToggleDetails;
+        if (toggleMaxLimit) {
+          if (toggleMaxLimit >= toggledCount) {
+            toggleLimit = toggleMaxLimit - toggledCount;
+          } else {
+            //We can end up here if we decide to reduce max limit value in firestore
+            toggleLimit = 0;
           }
-          setMaxToggleLimit(toggleMaxLimit);
         }
-        setShouldShowLoader(false);
+        maxToggleLimit = toggleMaxLimit;
       }
-    })();
-  }, [shouldFetchTokenDetails, getTokenToggleDetails]);
-
-  return {shouldShowLoader, toggleLimit, maxToggleLimit};
+      return {toggleLimit, maxToggleLimit};
+    },
+    staleTime: ONE_MINUTE,
+    cacheTime: ONE_MINUTE,
+  });
+  return {
+    ...queryResult,
+    toggleLimit: queryResult?.data?.toggleLimit,
+    maxToggleLimit: queryResult?.data?.maxToggleLimit,
+  }
```

Comment:
We shouldn't add these two top level `toggleLimit` and `maxToggleLimit` in the response. The return format of react-query is (or at least will be) well known for us after using it. Instead suffix the hook name with 'query', so its named `useTokenToggleDetailsQuery`, and we will know its a react-query and to look into the `data` field for the return values.